docker-app/qfieldcloud/core/views/projects_views.py

```diff
@@ -115,6 +116,33 @@ def get_queryset(self):
             )
         return projects
 
+    def perform_update(self, serializer):
+        # Here we do an additional check if the owner has changed. If so, the reciever
+        # of the project must have enough storage quota, otherwise the transfer is
+        # not permitted.
+
+        # TODO: this should be moved to some more reusable place. Maybe in Project.clean() ?
+        # But then it would also be enforced by admin, which we don't want I guess...
+
+        with transaction.atomic():
+            old_owner = serializer.instance.owner
+            super().perform_update(serializer)
+            new_owner = serializer.instance.owner
+
+            # If owner has not changed, no additional check is made
+            if old_owner == new_owner:
+                return
```

Comment:
I tried to thing if we can check this before opening the transaction, but failed to find a solution. Reminder just in case you have an idea to offload postgres bit.

Reply:
We could compare username in the json to the username of the instance, but not sure this is robust before validation (e.g. it could mix upper/lowercases, so that old_ownername != new_owndername, even if they represent the same object). And if we were to validate it before, we'd hit the DB anyway.

In any case, this doesn't happen too frequently, IMO not making a difference for the DB ? But I don't mind too much, I can change to the above idea if you prefer avoiding transaction (indeed a bit counterinutitive to do a change and rollback afterwards).

I switched to decorator syntax though.

Reply:
It is good, thanks!